src/main.ts

```diff
@@ -138,7 +134,7 @@ export const compression = (
 
       contentType = resContentType ? resContentType : 'text/plain'
 
-      const buffer = await res.arrayBuffer()
+      const buffer = await res.clone().arrayBuffer()
```

Comment:
This is the actual fix. The other changes are tests and minor cleanup of issues found by my IDE.